spec/controllers/products_controller_spec.rb

```diff
@@ -0,0 +1,104 @@
+# frozen_string_literal: true
+
+require 'rails_helper'
+
+RSpec.describe ProductsController, type: :controller do
+  let(:product) { Product.create(name: 'Product', image: 'product.jpg', code: '12345RE', price: 1000, description: 'My Product') }
```

Comment:
move product creation to factories, use factory_bot and faker gems.
Do the same for products with bad data using traits